src/Refactoring-Core/RBDeprecateMethodRefactoring.class.st

```diff
@@ -80,7 +80,12 @@ RBDeprecateMethodRefactoring class >> model: aRBSmalltalk deprecateMethod: aSele
 { #category : #preconditions }
 RBDeprecateMethodRefactoring >> applicabilityPreconditions [
 
-	^ RBCondition definesSelector: oldSelector in: class
+	^ (RBCondition
+		   withBlock: [
+		   newSelector isEmptyOrNil or: [ newSelector = oldSelector ] ]
+		   errorString:
+		   'The new selectors cannot be an empty string or same as the old selector.') not
```

Comment:
Instead of using "or:" wouldn't it be better to have two conditions so that the user knows exactly what was the problem?

Reply:
Yes it would! However, because of the way the `RBCondition` is implemented it's hard to produce the change you mentioned. This is on my todo but requires a lot of time since whole RBCondition needs to be refactored.

This is the pop-up right now:
<img width="657" alt="Screenshot 2023-08-17 at 14 51 04" src="https://github.com/pharo-project/pharo/assets/34557616/aab5c724-9440-445b-88e3-4ecf3981d19d">

This is the pop-up when I split condition on two smaller ones:
<img width="823" alt="Screenshot 2023-08-17 at 14 52 25" src="https://github.com/pharo-project/pharo/assets/34557616/a01692e7-a446-4fbf-82ee-6b057897b317">

The OR in between is added by the RBCondition. This is something that I want to rework since the very beginning, but unfortunately is not high on priority list... :(

Reply:
Oh I see! 
This is indeed not user friendly to merge all the warnings when we could just select the failing checks and display only those.